analyzer/src/main/java/org/jboss/gm/analyzer/alignment/io/SettingsFileIO.java

```diff
@@ -78,6 +79,43 @@ public String writeProjectNameIfNeeded(File rootDir) throws IOException {
         return result;
     }
 
+    /**
+     * If the DokkaVersion is less than or equal to 0.9.18 then this should
+     * inject the appropriate resolutionStrategy into the settings.gradle
+     *
+     * @param rootProject the root directory
+     * @param dokkaVersion the Dokka plugin version
+     * @throws IOException if an error occurs
+     */
+    public void writeDokkaSettings(File rootProject, PluginUtils.DokkaVersion dokkaVersion)
+            throws IOException {
+        if (dokkaVersion != PluginUtils.DokkaVersion.MINIMUM) {
+            return;
+        }
+        for (String s : Arrays.asList("settings.gradle", "settings.gradle.kts")) {
+            File settings = new File(rootProject, s);
+            if (settings.exists()) {
+                String settingsContents = FileUtils.readFileToString(settings, Charset.defaultCharset());
+                logger.info("Updating {} with Dokka resolutionStrategy information", settings);
+
+                if (settingsContents.contains("resolutionStrategy")) {
+                    // Existing strategy so inject another
+                    settingsContents = settingsContents.replaceFirst("(?s)resolutionStrategy(\\s|\\n|$)+\\{",
+                            "resolutionStrategy {\n eachPlugin { if (requested.id.id == \"org.jetbrains.dokka\") { useVersion(\"0.9.18\") } }\n");
+                } else if (settingsContents.contains("pluginManagement")) {
+                    // Existing pluginManagement but no strategy so inject that
+                    settingsContents = settingsContents.replaceFirst("(?s)pluginManagement(\\s|\\n|$)+\\{",
```

Comment:
I think that `\\s` is already matching newlines (`\\n`). Also, then it should handle a case we probably won't hit in practice of a `\\r?\\n`.

Reply:
This is the same pattern (I think) that I used successfully in PluginUtils.

Reply:
All I am saying is `\\s` already is "A whitespace character: `[ \t\n\x0B\f\r]`".

Reply:
Ah I see, thanks I hadn't realised whitespace included newline. I'll change both for consistency.